src/common/proxy_wasm/ngx_proxy_wasm_properties.c

```diff
@@ -935,16 +945,29 @@ ngx_proxy_wasm_properties_set_host(ngx_proxy_wasm_ctx_t *pwctx,
         ngx_memcpy(hpn->sn.str.data, path->data, path->len);
     }
 
-    hpn->value.len = value->len;
-    hpn->value.data = ngx_pstrdup(pwctx->pool, value);
-    if (hpn->value.data == NULL) {
-        return NGX_ERROR;
+    if (value->data == NULL) {
```

Comment:
> Can this branch be taken when hpn == NULL?

No, because the `else` branch in the test above creates a new `hpn` if it doesn't exist. The flow ended up a bit tricky because of the `is_const` and `!is_const` cases.

The iteration you reviewed also contained one inefficiency (if you tried to `set_property` to `NULL` and there wasn't one there before, it would store one non-const entry).

Fixing that added yet another `if` case, so I ended up reworking the whole thing to make it look more linear, which made the code a bit longer but hopefully easier to follow and just as efficient.